src/main/java/org/jenkinsci/plugins/oic/properties/EscapeHatch.java

```diff
@@ -0,0 +1,155 @@
+package org.jenkinsci.plugins.oic.properties;
+
+import static org.apache.commons.lang.StringUtils.isNotBlank;
+
+import edu.umd.cs.findbugs.annotations.CheckForNull;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import hudson.Extension;
+import hudson.Util;
+import hudson.security.SecurityRealm;
+import hudson.security.csrf.CrumbExclusion;
+import hudson.util.Secret;
+import jakarta.servlet.FilterChain;
+import jakarta.servlet.ServletException;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.Random;
+import java.util.regex.Pattern;
+import jenkins.security.FIPS140;
+import jenkins.security.SecurityListener;
+import org.jenkinsci.plugins.oic.OicProperty;
+import org.jenkinsci.plugins.oic.OicPropertyDescriptor;
+import org.jenkinsci.plugins.oic.OicUserDetails;
+import org.kohsuke.stapler.DataBoundConstructor;
+import org.springframework.security.authentication.BadCredentialsException;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.GrantedAuthority;
+import org.springframework.security.core.authority.SimpleGrantedAuthority;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.crypto.bcrypt.BCrypt;
+
+/**
+ * Escape hatch for authentication, allowing users to log in with a username and password.
+ * This is intended for emergency access and should be used with caution.
+ */
+public class EscapeHatch extends OicProperty {
+    public static final Pattern B_CRYPT_PATTERN = Pattern.compile("\\A\\$[^$]+\\$\\d+\\$[./0-9A-Za-z]{53}");
+
+    @CheckForNull
+    private final String username;
+
+    @CheckForNull
+    private final String group;
+
+    @NonNull
+    private final Secret secret;
+
+    @DataBoundConstructor
+    public EscapeHatch(@NonNull String username, @CheckForNull String group, @NonNull Secret secret) {
+        if (FIPS140.useCompliantAlgorithms()) {
+            throw new IllegalStateException("Cannot use Escape Hatch in FIPS-140 mode");
+        }
+        this.username = Util.fixEmptyAndTrim(username);
```

Comment:
username is required to work correctly - throw a `Descriptor.FormException` if it is empty or blank?